pkg/cli/init.go

```diff
@@ -71,39 +70,18 @@ func InitConfig() {
 	}
 }
 
-// SetupPrometheus sets up the prometheus server
+// SetupPrometheus sets up the Prometheus server, registering its handler on http.DefaultServeMux
 func SetupPrometheus(ctx context.Context, logger *zap.SugaredLogger, name string) (metrics.MetricCollector, error) {
-	if name == "" {
-		return nil, errors.New("name cannot be empty")
-	}
-	m := metrics.FromContext(ctx, name)
 	enablePrometheus := viper.GetBool("enable-prometheus")
-	prometheusPort := viper.GetInt("prometheus-addr")
 	if !enablePrometheus {
 		return nil, nil
 	}
 
-	go func() {
-		http.Handle("/metrics", m.MetricsHandler())
-		logger.Infof("Prometheus server is listening on: %d", prometheusPort)
-		server := &http.Server{Addr: fmt.Sprintf(":%d", prometheusPort)}
-
-		// Start server in a goroutine so that it doesn't block
-		go func() {
-			if err := server.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
```

Comment:
Previously, if my understanding is correct (this is still quite new to me but I tried to test and read the `net/http` docs closely and I believe I am correct) we were not ever serving `/metrics` separately.

What we would do is serve `http.DefaultServeMux` off of two separate ports: on being the GQL listen port and the other being the Prometheus port. Both of these ports would listen on the normal GQL path (which I think is `/query` but haven't actually tested) and `/metrics`, and if you had a playground being served then it would be served off `/`.

So I don't think the code made much sense before. I tried to clarify this in the commit message but if you want me to change anything let me know. It is also possible I have made a mistake.

Reply:
ah ok, so it was not needed in the first place. Good catch.